src/lens/unfollow.js

```diff
@@ -0,0 +1,80 @@
+import { gql } from '@apollo/client/core';
+import { ethers } from 'ethers';
+import { apolloClient } from '../apollo-client';
+import { LENS_HUB_ABI } from '@/lens/utils/config-abi';
+import { getAddress, getSigner, signedTypeData, splitSignature } from '@/ethers-service';
+
+const CREATE_UNFOLLOW_TYPED_DATA = `
+  mutation($request: UnfollowRequest!) { 
+    createUnfollowTypedData(request: $request) {
+      id
+      expiresAt
+      typedData {
+        domain {
+          name
+          chainId
+          version
+          verifyingContract
+        }
+        types {
+          BurnWithSig {
+            name
+            type
+          }
+        }
+        value {
+          nonce
+          deadline
+          tokenId
+        }
+      }
+    }
+ }
+`;
+
+const createUnfollowTypedData = (profile) => {
+  return apolloClient.mutate({
+    mutation: gql(CREATE_UNFOLLOW_TYPED_DATA),
+    variables: {
+      request: {
+        profile,
+      },
+    },
+  });
+};
+
+const unfollow = async (unfollowProfileId) => {
+  const address = await getAddress();
+  console.log('unfollow: address', address);
```

Comment:
Once you're done, could you clean up the console.log please ?
It's quite noisy for the one not working on the feature.